src/ContainerInstance.ts

```diff
@@ -350,6 +351,9 @@ export class ContainerInstance {
      */
     private initializeParams(type: Function, paramTypes: any[]): any[] {
         return paramTypes.map((paramType, index) => {
+            if (paramType === undefined)
```

Comment:
I can merge it, but I really afraid if there is a case where paramType can be undefined and we'll have this error unexpected.

Reply:
Ok, I will add some extra unit tests to cover some scenarios but I don't think that there are really any legit cases where a `Type` parameter should be undefined. These are coming from auto-generated metadata from typescript and it should not let you have undefined types, this should only occur in the case where a circular import is happening (which I'll try to prove with some tests) and only when trying to resolve a constructor.

Reply:
But also, even though I think it shouldn't have any automatically adverse affects on people this is technically a breaking change and I would include this in a Major version bump so that in case there are issues for people it isn't automatically thrust upon them.